TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -430,57 +485,39 @@ private void SlotResultHandler(int[] betAndSlotResults)
                             winnings = betAmount * ONE_ONE_ANY_PAYOUT_MULTIPLIER;
                         else
                             winnings = betAmount * ONE_ANY_ANY_PAYOUT_MULTIPLIER;
                         break;
                     }
                 default: // losing roll, winnings remain 0
+                    winnings = 0;
                     break;
             }
-            if (winnings > 0)
-            {
-                // dispatch event to tell object the winning amount
-                userClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.SetWinAmount, new short[1] { Convert.ToInt16(winnings) }));
-                // dispatch event to tell the object that the user has won
-                userClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ExecuteWin, new short[1] { betAmount }));
-            }
-            else
-            {
-                // dispatch event to tell object that the user has lost
-                userClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ExecuteLoss, new short[1] { betAmount }));
-            }
-
-            // send winnings and spins results to UI
-            userClient.Send("slots_spin", new Byte[] { Convert.ToByte(winnings / 255), Convert.ToByte(winnings % 255),
-                (byte)slotsStops[betAndSlotResults[1]], (byte)slotsStops[betAndSlotResults[2]], (byte)slotsStops[betAndSlotResults[3]] });
-            State = VMEODSlotsStates.Running;
+            return winnings;
         }
         private void SlotsCloseHandler(string evt, string arg, VMEODClient client)
         {
             Server.Disconnect(client);
         }
         private int[] RollNewGame(int betAmount)
         {
-            int wheelOneStop = slotsRandom.Next(totalStops);
-            int wheelTwoStop = slotsRandom.Next(totalStops);
-            int wheelThreeStop = slotsRandom.Next(totalStops);
+            int wheelOneStop = SlotsRandom.Next(TotalStops);
+            int wheelTwoStop = SlotsRandom.Next(TotalStops);
+            int wheelThreeStop = SlotsRandom.Next(TotalStops);
             return new int[4] { betAmount, wheelOneStop, wheelTwoStop, wheelThreeStop};
         }
-        private void WheelsStoppedHandler(string evt, string payoutString, VMEODClient client)
+        private void WheelsStoppedHandler(string evt, string uselessString, VMEODClient client)
         {
-            if (payoutString[0] != '0')
+            if (CurrentWinnings > 0)
             {
-                State = VMEODSlotsStates.DisplayingWin;
                 // send win event with random number for message displayed
-                client.Send("slots_display_win", "" + slotsRandom.Next(25, 30) + "%" + payoutString);
+                client.Send("slots_display_win", "" + SlotsRandom.Next(25, 30) + "%" + CurrentWinnings);
 
                 // pay the player
-                short winnings = Int16.Parse(payoutString);
-                
                 var VM = client.vm;
```

Comment:
Set current winnings to 0 immediately before calling this, and use a copy instead. Since there is a delay between a transaction starting and completing, current winnings will still be greater than 0 for a split second after this handler is run, so the handler could be run again to get a double payout.